plugins/datatools/include/datatools/io/dataformat/ImageCalls.h

```diff
@@ -0,0 +1,275 @@
+#pragma once
+
+#include "DataFormat.h"
+#include "ImageElementType.h"
+
+namespace megamol {
+namespace datatools {
+namespace io {
+namespace dataformat {
+
+// do we want to pull out an interface? what kind of code do we write against this?
+// which variants do we want to handle dynamically? different channel types probably. dimensionality is unlikely.
+// different numbers of components?
+
+// TODO: get rid of NumComponents. There is only one, keep multiple images if you want more channels.
+// exception: RGBA8 as UINT32 ? Does it matter?
+
+// typename ChannelType does not work, that is not a static decision.
+// For some data sources you only know after actually opening the file what's in it.
+// TODO: volume ist ein byte-vektor und wir benutzen span. fertig.
+template<uint8_t Dimensions>
```

Comment:
Do we need dimension as static information since we do not change available methods?

Reply:
I was wondering whether we should disable some of the methods depending on Dimensions. I am really not sure whether we should just treat dims dynamically too. A transfer function is a perfectly OK volume, strictly speaking. It's a bit thin, but not broken or something...

Reply:
With the dimension discerning types, we have either three statically different call transporting 1D, 2D, or 3D volumes, or yet another dispatch. I would argue for dimension as template parameter, if we have a significant gain from it.